### DISCUSSION TITLE
ROCm 6.1.2 Release

### DISCUSSION THREAD

**Discussion by samjwu** [ROCm/ROCm#3235] [Announcements]
# ROCm 6.1.2 release notes

ROCm 6.1.2 includes enhancements to SMI tools and improvements to some libraries.

### OS support

ROCm 6.1.2 has been tested against a pre-release version of Ubuntu 22.04.5 (kernel: 5.15 [GA], 6.8 [HWE]).

### AMD SMI

AMD SMI for ROCm 6.1.2

#### Additions

* Added process isolation and clean shader APIs and CLI commands.
  * `amdsmi_get_gpu_process_isolation()`
  * `amdsmi_set_gpu_process_isolation()`
  * `amdsmi_set_gpu_clear_sram_data()`
* Added the `MIN_POWER` metric to output provided by `amd-smi static --limit`.

#### Optimizations

* Updated the `amd-smi monitor --pcie` output to prevent delays with the `monitor` command.

#### Changes

* Updated `amismi_get_power_cap_info` to return values in uW instead of W.
* Updated Python library return types for `amdsmi_get_gpu_memory_reserved_pages` and `amdsmi_get_gpu_bad_page_info`.
* Updated the output of `amd-smi metric --ecc-blocks` to show counters available from blocks.

#### Fixes

* `amdsmi_get_gpu_board_info()` no longer returns junk character strings.
* `amd-smi metric --power` now correctly details power output for RDNA3, RDNA2, and MI1x devices.
* Fixed the `amdsmitstReadWrite.TestPowerCapReadWrite` test for RDNA3, RDNA2, and MI100 devices.
* Fixed an issue with the `amdsmi_get_gpu_memory_reserved_pages` and `amdsmi_get_gpu_bad_page_info` Python interface calls.

#### Removals

* Removed the `amdsmi_get_gpu_process_info` API from the Python library. It was removed from the C library in an earlier release.

See the AMD SMI [detailed changelog](https://github.com/ROCm/amdsmi/blob/rocm-6.1.x/CHANGELOG.md) with code samples for more information.

### ROCm SMI

ROCm SMI for ROCm 6.1.2

#### Additions

* Added the ring hang event to the `amdsmi_evt_notification_type_t` enum.

#### Fixes

* Fixed an issue causing ROCm SMI to incorrectly report GPU utilization for RDNA3 GPUs. See the issue on [GitHub](https://github.com/ROCm/ROCm/issues/3112).
* Fixed the parsing of `pp_od_clk_voltage` in `get_od_clk_volt_info` to work better with MI-series hardware.

### RCCL

RCCL 2.18.6 for ROCm 6.1.2

#### Changes

* Reduced `NCCL_TOPO_MAX_NODES` to limit stack usage and avoid stack overflow.

### rocBLAS

rocBLAS 4.1.2 for ROCm 6.1.2

#### Optimizations

* Tuned BBS TN and TT operations on the CDNA3 architecture.

#### Fixes

* Fixed an issue related to obtaining solutions for BF16 TT operations.

### rocDecode

rocDecode 0.6.0 for ROCm 6.1.2

#### Additions

* Added support for FFmpeg v5.x.

#### Optimizations

* Updated error checking in the `rocDecode-setup.py` script.

#### Changes

* Updated core dependencies.
* Updated to support the use of public LibVA headers.

#### Fixes

* Fixed some package dependencies.

## Upcoming changes

* A future release will enable the use of HIPCC compiled binaries `hipcc.bin` and `hipconfig.bin` by default. No action is needed by users; you may continue calling high-level Perl scripts `hipcc` and `hipconfig`. `hipcc.bin` and `hipconfig.bin` will be invoked by the high-level Perl scripts. To revert to the previous behavior and invoke `hipcc.pl` and `hipconfig.pl`, set the `HIP_USE_PERL_SCRIPTS` environment variable to `1`.
* A subsequent release will remove high-level HIPCC Perl scripts from `hipcc` and `hipconfig`. This release will remove the `HIP_USE_PERL_SCRIPTS` environment variable. It will rename `hipcc.bin` and `hipconfig.bin` to `hipcc` and `hipconfig` respectively. No action is needed by the users. To revert to the previous behavior, invoke `hipcc.pl` and `hipconfig.pl` explicitly.
* A subsequent release will remove `hipcc.pl` and `hipconfig.pl` for HIPCC.

<hr /><em>This discussion was created from the release <a href='https://github.com/ROCm/ROCm/releases/tag/rocm-6.1.2'>ROCm 6.1.2 Release</a>.</em>

**Comment by FeepingCreature** [comment]
Any news on better MES hang recovery or even fixes?

**Comment by lamikr** [comment]
rocm 6.1.3 seems to be released but rocm-6.1.3 git tags are missing.
